main/xorgproto/APKBUILD

```diff
@@ -0,0 +1,73 @@
+# Contributor: prspkt <prspkt@protonmail.com>
+# Maintainer: prspkt <prspkt@protonmail.com>
+pkgname=xorgproto
+pkgver=2018.4
+pkgrel=0
+pkgdesc="Combined X.Org X11 protocol headers"
+url="https://xorg.freedesktop.org"
+arch="noarch"
+license="MIT"
+makedepends="meson util-macros"
+replaces="bigreqsproto compositeproto damageproto dri2proto dri3proto fixesproto
+	fontsproto glproto inputproto kbproto presentproto printproto randrproto
+	recordproto renderproto resourceproto scrnsaverproto videoproto xcmiscproto
+	xextproto xf86bigfontproto xf86dgaproto xf86driproto xf86miscproto
+	xf86vidmodeproto xineramaproto xproto
+	"
+provides="bigreqsproto=$pkgver-r$pkgrel
+	compositeproto=$pkgver-r$pkgrel
+	damageproto=$pkgver-r$pkgrel
+	dri2proto=$pkgver-r$pkgrel
+	dri3proto=$pkgver-r$pkgrel
+	fixesproto=$pkgver-r$pkgrel
+	fontsproto=$pkgver-r$pkgrel
+	glproto=$pkgver-r$pkgrel
+	inputproto=$pkgver-r$pkgrel
+	kbproto=$pkgver-r$pkgrel
+	presentproto=$pkgver-r$pkgrel
+	printproto=$pkgver-r$pkgrel
+	randrproto=$pkgver-r$pkgrel
+	recordproto=$pkgver-r$pkgrel
+	renderproto=$pkgver-r$pkgrel
+	resourceproto=$pkgver-r$pkgrel
+	scrnsaverproto=$pkgver-r$pkgrel
+	videoproto=$pkgver-r$pkgrel
+	xcmiscproto=$pkgver-r$pkgrel
+	xextproto=$pkgver-r$pkgrel
+	xf86bigfontproto=$pkgver-r$pkgrel
+	xf86dgaproto=$pkgver-r$pkgrel
+	xf86driproto=$pkgver-r$pkgrel
+	xf86miscproto=$pkgver-r$pkgrel
+	xf86vidmodeproto=$pkgver-r$pkgrel
+	xineramaproto=$pkgver-r$pkgrel
+	xproto=$pkgver-r$pkgrel
+	"
```

Comment:
This will cause problems because the build order calculation does not consider provides. We should clean up all the packages that has `*proto` in makedepends.

Reply:
Yes. I will start the cleanup and submit a PR.

Reply:
@ncopa If you know that it will cause problems, then why did you merge it (before the fix is provided)?!

Reply:
Because it only causes problems build time, to detect the build order, so it will likely not be a problem unless you do a full rebuild, which we do for v3.9. Even if someone get bitten by it before that, it is trivial to fix when you are affected.

The fix affects many packages (~70) so I thought it will be a challenge to do everything in one go without introducing conflicts with other pull requests. So i though we can clean up as we go.